src/audit/unpinned_uses.rs

```diff
@@ -21,15 +21,26 @@ impl WorkflowAudit for UnpinnedUses {
             return Ok(vec![]);
         };
 
-        if uses.unpinned() {
+        let (annotation, severity) = if uses.unpinned() {
+            (
+                "action is not pinned to a tag, branch, or hash ref",
+                Severity::Medium,
+            )
+        } else if uses.unhashed() {
+            ("action is not pinned to a hash ref", Severity::Low)
+        } else {
+            ("", Severity::Informational)
+        };
```

Comment:
Doing a `return` in the `else` case here should avoid the need for an "empty" state:

```suggestion
        let (annotation, severity) = if uses.unpinned() {
            (
                "action is not pinned to a tag, branch, or hash ref",
                Severity::Medium,
            )
        } else if uses.unhashed() {
            ("action is not pinned to a hash ref", Severity::Low)
        } else {
            return Ok(vec![]);
        };
```